docs/adr/0002-yarn-v4.md

```diff
@@ -0,0 +1,243 @@
+➤ YN0000: ┌ Resolution step
+➤ YN0085: │ + @types/lodash@npm:4.17.13, lodash@npm:4.17.21
+➤ YN0000: └ Completed
+➤ YN0000: ┌ Fetch step
+➤ YN0013: │ A package was added to the project (+ 957.26 KiB).
+➤ YN0000: └ Completed in 0s 252ms
+➤ YN0000: ┌ Link step
+➤ YN0000: └ Completed
+➤ YN0000: · Done in 0s 313ms
+
+$ cat package.json 
+{
+  "name": "yarn-types",
+  "packageManager": "yarn@4.5.3",
+  "dependencies": {
+    "lodash": "^4.17.21"
+  },
+  "devDependencies": {
+    "@types/lodash": "^4"
+  }
+}
+```
+</details>
+
+### pnpDataPath is no longer configurable
+
+This used to be a configurable key in the `.yarnrc.yml` file. The previous default path, `./.pnp.data.json`, is now hard-coded and can't be changed.
+
+The only mention to `pnpDataPath` in Cachi2 is the [check](https://github.com/containerbuildsystem/cachi2/blob/a5f19c6f9be90be4289beee35ecccd2827bbb328/cachi2/core/package_managers/yarn/main.py#L43) for paths pointing outside of the repo. The check can be kept the same way so Yarn v3 can still be covered, and it will simply be skipped in v4.
+
+### Yarn now caches npm version metadata
+
+Although Yarn only seems to create the cache folder (`{globalFolder}/metadata/npm`) when the [hardened mode](https://yarnpkg.com/configuration/yarnrc#enableHardenedMode) is enabled. Simply running `yarn install` on a local project or even prefetching with Cachi2 doesn't seem to generate this extra folder.
+
+The metadata, in itself, is a collection of json files with a few kilobytes each. In case we decide to enable the hardened mode, it'd be ideal to delete this npm metadata this from the output folder, although there's no harm in keeping it.
+
+### Changes in yarn.lock
+
+When updating v3 projects to v4, some differences in the lockfile appear. These are some picks from updating the [disallowed-protocols]() branch on the cachi2-yarn-berry test project:
+
+**`npm:` locator added to npm dependencies:**
+```
+   dependencies:
+-    chownr: ^2.0.0
+-    fs-minipass: ^2.0.0
+-    minipass: ^5.0.0
+-    minizlib: ^2.1.1
+-    mkdirp: ^1.0.3
+-    yallist: ^4.0.0
+-  checksum: ...
++    chownr: "npm:^2.0.0"
++    fs-minipass: "npm:^2.0.0"
++    minipass: "npm:^5.0.0"
++    minizlib: "npm:^2.1.1"
++    mkdirp: "npm:^1.0.3"
++    yallist: "npm:^4.0.0"
++  checksum: ...
+```
+
+**Changes to some instances of the file locator:**
+```
+-  resolution: "strip-ansi-tarball@file:external-packages/strip-ansi-4.0.0.tgz::locator=berryscary%40workspace%3A."
++  resolution: "strip-ansi-tarball@file:external-packages/strip-ansi-4.0.0.tgz#external-packages/strip-ansi-4.0.0.tgz::hash=e17689&locator=berryscary%40workspace%3A."
+```
+
+Yarn v4 introduced the subdirectory (`#external-packages/strip-ansi-4.0.0.tgz`) and the hash (`::hash=e17689`), which are already [handled](https://github.com/containerbuildsystem/cachi2/blob/a5f19c6f9be90be4289beee35ecccd2827bbb328/cachi2/core/package_managers/yarn/locators.py#L77-L79) by the v3 implementation in Cachi2.
```

Comment:
So, does this mean that no change was introduced after all?

Reply:
The lockfile was modified, but Cachi2 is already able to parse the changes. In the end, there was no effective change from Cachi2's perspective, since the prefetch will still be performed the same way.